### PR TITLE
unit test for kubectl config set-cluster

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:
unit test for create cluster

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
i want test secure mode,but CA path how set?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
@adohe

**Comment by adohe** [comment]
@zjj2wry I would review this today, sorry about late response.

**Comment by zjj2wry** [comment]
@adohe ok,think you

File: pkg/kubectl/cmd/config/create_cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+func TestModifyCluster(t *testing.T) {
+	conf := clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"my-cluster": {Server: "https://192.168.0.1"},
+		},
+	}
+	test := createClusterTest{
+		description: "Testing kubectl config set-cluster a exist cluster",
```
Comment (adohe) [inline pkg/kubectl/cmd/config/create_cluster_test.go]: confused description.

**Comment by adohe** [comment]
@zjj2wry I am sorry about the late, I was just too busy recently. This generally lgtm. Could you please rebase this? Then I will apply the label.

**Comment by zjj2wry** [comment]
@adohe OK,take care of yourself  ：）

**Comment by adohe** [comment]
@k8s-bot ok to test

**Comment by zjj2wry** [comment]
@k8s-bot verify test this

**Comment by zjj2wry** [comment]
@k8s-bot verify test this

**Comment by zjj2wry** [comment]
@k8s-bot verify test this

**Comment by adohe** [comment]
@k8s-bot verify test this

**Comment by zjj2wry** [comment]
@k8s-bot verify test this

**Comment by zjj2wry** [comment]
@thockin i rebase lastest master. but the **do-not-merge** tag always exists. how can i remove it ? think you very much.

**Comment by thockin** [comment]
I cleared do-not-merge, but someone has to LGTM who knows this code.

**Comment by zjj2wry** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by zjj2wry** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by zjj2wry** [comment]
@k8s-bot verify test this

**Comment by zjj2wry** [comment]
@fabianofranz sorry about ping you often. can you help me review this.the assign reviewer maybe not online recent. think you ~

File: pkg/kubectl/cmd/config/create_cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	args           []string
+	flags          []string
+	expected       string
+	expectedConfig clientcmdapi.Config
+}
+
+func TestCreateCluster(t *testing.T) {
+	conf := clientcmdapi.Config{}
+	test := createClusterTest{
+		description: "Testing kubectl config set-cluster a cluster",
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/config/create_cluster_test.go]: Make it "Testing 'kubectl config set-cluster' with a new cluster"

File: pkg/kubectl/cmd/config/create_cluster_test.go (hunk 1 above)
Comment (fabianofranz) [inline pkg/kubectl/cmd/config/create_cluster_test.go]: Make it "Testing 'kubectl config set-cluster' with an existing cluster"

File: pkg/kubectl/cmd/config/create_cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	pathOptions := clientcmd.NewDefaultPathOptions()
+	pathOptions.GlobalFile = fakeKubeFile.Name()
+	pathOptions.EnvVar = ""
+	buf := bytes.NewBuffer([]byte{})
+	cmd := NewCmdConfigSetCluster(buf, pathOptions)
+	cmd.SetArgs(test.args)
+	cmd.Flags().Parse(test.flags)
+	if err := cmd.Execute(); err != nil {
+		t.Fatalf("unexpected error executing command: %v,kubectl set-cluster args:%v,flags:%v", err, test.args, test.flags)
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/config/create_cluster_test.go]:
```
t.Fatalf("unexpected error executing command: %v, args: %v, flags: %v", err, test.args, test.flags)
```

File: pkg/kubectl/cmd/config/create_cluster_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	config, err := clientcmd.LoadFromFile(fakeKubeFile.Name())
+	if err != nil {
+		t.Fatalf("unexpected error loading kubeconfig file: %v", err)
+	}
+	if len(test.expected) != 0 {
+		if buf.String() != test.expected {
+			t.Errorf("Failded in %q\n expected %v\n but got %v", test.description, test.expected, buf.String())
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/config/create_cluster_test.go]: "Failed ..."

File: pkg/kubectl/cmd/config/create_cluster_test.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		}
+	}
+	if len(test.args) > 0 {
+		cluster, ok := config.Clusters[test.args[0]]
+		if !ok {
+			t.Errorf("expected create cluster %v, but got nil", test.args[0])
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/config/create_cluster_test.go]:
```
t.Errorf("expected cluster %v, got nil", test.args[0])
```

**Comment by fabianofranz** [comment]
@zjj2wry just a few minor nits.

**Comment by zjj2wry** [comment]
@fabianofranz thank you for review :)

**Comment by fabianofranz** [comment]
@zjj2wry thank you!

/lgtm

**Comment by zjj2wry** [comment]
@k8s-bot pull-kubernetes-node-e2e test this
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this
@k8s-bot pull-kubernetes-e2e-kops-aws test this
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**